Support use of colorama up to 0.3.9 (closes #2666)

zyv opened the pull request:
Colorama bugfix release 0.3.9 is available and contains no incompatible changes. There is no need to restrict use to less or equal 0.3.7. Supersedes PR #2666.

zyv commented:
/cc @JordonPhillips @stealthycoin @rjschwei

zyv commented:
Also addresses #2610.

haylo75 commented:
I just came upon this open PR while researching a colorama version conflict in our local dev environment.  I see setup.py now has merge conflicts.  Any ETA on merging this PR?

zyv commented:
I've fixed the conflicts yet again... Hey, Amazonians, this can't be that hard to merge and will help a lot of people! Could you please consider my PR?

zyv commented:
Happy New Year 2018!

zyv commented:
One day...

haylo75 commented:
@zyv thanks for keeping this PR up to date.  Yes, one day...

onedeveloper commented:
What is the status for this? any chance this will get merged?

holybit commented:
Fix it, please, ASAP. For weeks now I am seeing `pip3 install --upgrade awscli` warnings - getting old ... fast.

zyv commented:
@joguSD be a hero, merge this oneliner! :)

joguSD commented:
It looks like `colorama` dropped support for Python 2.6 and 3.3 with version 0.3.9. This means increasing the upper bound will be relatively high friction for us. Additionally, this upper bound would need to be changed in `setup.cfg` and `requirements.txt` as well.

zyv commented:
@joguSD thank you for the pointer, I have adjusted the other files as well. Please note that I have updated the dependency to the latest released `colorama` version `0.3.9` back in April, 2017. The commit dropping Python 2.6 / 3.3 that you are worried about came in December, 2017 and is **not** included in this release!

https://github.com/tartley/colorama/commit/34f410b10e206546848eb7fdfedd0a33c9d57eb9

joguSD commented:
Just out of curiosity, what's the motivating factor to upgrade? Is there a bug fix or new feature in these later versions that you want?

rjschwei commented:
From my point of view:

2 issues:

- For a "from source" install there is clearly the clash of installing this source and some other source that needs a newer colorama. If both Python packages subscribe their dependency with "==" and one says "==x" and the other says "==y" then that's a problem. 
- From a packaging point of view one gets more or less the same problem if packagers use the version lock down that's set up in setup.py. Packagers do have the freedom to hide what's in setup.py as make up their own requerements such as ">="

joguSD commented:
@rjschwei Are there any packages in particular that require a newer version of colorama that are causing troubles for you?

browniebroke commented:
You can see the full list here, packages that require colorama>=0.3.9 in their setup.py:
https://libraries.io/pypi/colorama/usage?requirements=%3E%3D0.3.9

For me, the conflicting one is [aloe](https://libraries.io/github/aloetesting/aloe).

lhanekam commented:
@joguSD I work for one of the largest IT companies in the world and we have automation that targets both AWS and Azure. Deployment machines install both the AWS CLI and Azure SDK. The AWS CLI uses Colorama 0.3.7 and the Azure SDK uses Colorama 0.3.9. In addition, we use AWS S3 as the host for all deployment files including files for an Azure deployment. So, if we want to eliminate the incompatibility, we will need to create two separate virtual machines (one for AWS deployments and one for Azure deployments). In addition, we will no longer be able to use only AWS S3 for managing the deployment components for both AWS and Azure deployments. We would need to use Azure storage instead for Azure deployments, which I would prefer not to do. Version 0.3.9 was released over a year ago, so it would be very helpful if you could keep up with Microsoft on version changes. Thanks for your consideration.

trishankatdatadog commented:
@joguSD I understand you closed my duplicate PR, but is there a timeline for when this relatively simple PR will get merged?

trishankatdatadog commented:
Thanks very much @joguSD and @jamesls !

AlJohri commented:
@joguSD can we bump colorama again to `0.4.1` from November 2018? https://pypi.org/project/colorama/0.4.1/

I keep running into:

```
ERROR: awscli 1.16.209 has requirement colorama<=0.3.9,>=0.2.5, but you'll have colorama 0.4.1 which is incompatible.
```
